File: Swift/BOJ/1000/1806.swift
//
//  1806.swift
//  SwiftAlgorithm
//
//  Created by 유정주 on 2023/03/19.
//

import Foundation

let ns = readLine()!.components(separatedBy: " ").map { Int($0)! }
let n = ns[0], s = ns[1]

var input = readLine()!.components(separatedBy: " ").map { Int($0)! }

var left = 0, right = 0
var sum = 0
var result = Int.max
while left <= right {
    if sum >= s {
        result = min(result, right - left)
        sum -= input[left]
        left += 1
    } else {
        if right == n {
            break
        }
        sum += input[right]
        right += 1
    }
}

if result == Int.max {
    result = 0
}

print(result)

File: Python/BOJ/투포인터/1806.py
//
//  1806.py
//  Algorithm
//
//  Created by 유정주 on 2023/06/09.
//

import sys; readLine = sys.stdin.readline

n, s = map(int, readLine().split())
arr = list(map(int, readLine().split()))

result = 1e9
left = 0; right = 0
sum = arr[0]
while left <= right:
    if sum >= s:
        result = min(result, right - left + 1)
        sum -= arr[left]
        left += 1
    else:
        if right + 1 < n:
            right += 1
            sum += arr[right]
        else:
            break

if result == 1e9:
    print("0")
else:
    print(result)
